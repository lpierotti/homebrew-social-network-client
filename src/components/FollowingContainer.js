import React from 'react'
import { connect } from 'react-redux'
import { getFollows } from '../actions/users'
import FollowDisplay from './FollowDisplay'
import { Segment, Divider, Card } from 'semantic-ui-react'

class FollowingContainer extends React.Component {

	componentDidMount() {
		this.props.getFollows(this.props.id)
	}

	componentWillReceiveProps(nextProps) {
		if (this.props.id !== nextProps.id){
			this.props.getFollows(nextProps.id)
		}
	}

	render() {
		console.log(this.props)
		return (
			<div className={'followingContainer'}>
				<Segment className='container'>
					<h2>{this.props.currentUser === this.props.viewing ? 'Your' : `People ${this.props.viewing}` } Follows</h2>
					<Divider />
					<Card.Group itemsPerRow={4} textAlign='center'>
						{this.props.follows ? this.props.follows.map((follow, index) => <FollowDisplay key={index} data={follow}/>) : null}
					</Card.Group>
					<h2>Followers</h2>
					<Divider />
					<Card.Group itemsPerRow={4} textAlign='center'>
						{this.props.followers ? this.props.followers.map((follower, index) => <FollowDisplay key={index} data={follower}/>) : null}
					</Card.Group>
				</Segment>
			</div>
		)
	}
}

function mapStateToProps(state) {
	return {	
		follows: state.users.userFollowees,
		followers: state.users.userFollowers
	}
}

function mapDispatchToProps(dispatch) {
	return {
		getFollows: (id) => dispatch(getFollows(id))
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(FollowingContainer)