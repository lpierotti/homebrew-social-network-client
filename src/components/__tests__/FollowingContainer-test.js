import React from 'react';
import { mount } from 'enzyme';
import { FollowingContainer } from '../FollowingContainer';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom'

describe('Following container', () =>{
	it('renders correctly', () => {
		const tree = renderer.create(
			<FollowingContainer follows={[]} followers={[]} getFollows={() => {}}/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	})

	const mockGetFollowsfn = jest.fn()
	let wrapper;
	beforeAll(() => {
		let dummyFollows = [{id: 1, username: 'joe'}]
		let dummyFollowers = [{id: 2, username: 'bob'}]
		wrapper = mount(<StaticRouter context={{}}><FollowingContainer id={88} follows={dummyFollows} followers={dummyFollowers} getFollows={mockGetFollowsfn}/></StaticRouter>)
	})
	afterAll(() => {
		wrapper.unmount()
	})

	it('renders the div for the container', () => {
		expect(wrapper.find('.followingContainer').length).toBe(1)
	})

	it('renders the follows and followers headers', () => {
		expect(wrapper.find('h2').length).toBe(2)
	})

	describe('the getFollows function operates correctly', () => {
		it('calls the getFollows function when it is mounted', () => {
			expect(mockGetFollowsfn.mock.calls.length).toBe(1)
		})

		it('gets called with the correct argument', () => {
			expect(mockGetFollowsfn.mock.calls[0][0]).toBe(88)
		})

	})

	describe('the follows div', () => {

		it('successfully renders a follow display when there is a follow', () => {
			expect(wrapper.find('img').length).toBe(2)
			expect(wrapper.findWhere(n => n.type() === 'a' && n.text() === 'joe').exists()).toEqual(true)
		})
	})

	describe('the follower div', () => {

		it('successfully renders a follow display when there is a follower', () => {
			expect(wrapper.find('img').length).toBe(2)
			expect(wrapper.findWhere(n => n.type() === 'a' && n.text() === 'bob').exists()).toEqual(true)
		})
	})


})