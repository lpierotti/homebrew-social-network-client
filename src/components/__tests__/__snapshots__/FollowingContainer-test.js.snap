// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Following container renders correctly 1`] = `
<div
  className="followingContainer"
>
  <div
    className="ui segment"
    style={
      Object {
        "backgroundColor": "rgba(255,185,0,.5)",
      }
    }
  >
    <h2>
      Your
       Follows
    </h2>
    <div
      className="ui divider"
    />
    <div
      className="ui center aligned four cards"
    />
    <h2>
      Followers
    </h2>
    <div
      className="ui divider"
    />
    <div
      className="ui center aligned four cards"
    />
  </div>
</div>
`;
